osh/builtin.py

```diff
@@ -512,12 +513,35 @@ def __call__(self, arg_vec):
 
     names = arg_vec.strs[i:]
     if arg.n is not None:  # read a certain number of bytes
+      stdin = sys.stdin.fileno()
       try:
         name = names[0]
       except IndexError:
         name = 'REPLY'  # default variable name
-      s = posix.read(sys.stdin.fileno(), arg.n)
-      #log('read -n: %s = %s', name, s)
+      s = ""
+      if sys.stdin.isatty():  # set stdin to read in unbuffered mode
```

Comment:
I guess you can use stdin.isatty() instead of sys.stdin

Reply:
`stdin` is a number here, not an object. Calling `isatty` would through a reference error.